subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedTaskExecutionIntegrationTest.groovy

```diff
@@ -485,4 +485,34 @@ class CachedTaskExecutionIntegrationTest extends AbstractIntegrationSpec impleme
         then:
         result.assertTaskSkipped(":cacheable")
     }
+
+    def "downstream task stays cached when upstream task is loaded from cache"() {
```

Comment:
This fails if different snapshots are registered when loading from cache vs when snapshotting task outputs normally. I think we could make it more clear by cleaning only `compileTestJava`.

We should convert all the tests in `CachedTaskExecutionIntegrationTest` to use custom tasks at some point.